indico/queries/jobs.py

```diff
@@ -53,7 +52,7 @@ class JobStatus(RequestChain):
     Args:
         id (int): id of the job to query for status.
         wait (bool, optional): Wait for the job to complete? Default is True
-        timeout (float or int, optional): Timeout after this many seconds.
```

Comment:
Nice work on the fix + updating the tests!

Only feedback here - there's an impact to changing the interface/accepted types -- while you could submit an int for a float, if users are submitting floats and upgrade after we narrow the accepted types to int, they will have to change their scripts. 

If it's possible to accept floats here we should still. Thank you!